src/hera/workflows/task.py

```diff
@@ -71,22 +122,59 @@ def _get_dependency_tasks(self) -> List[str]:
     def _subtype(self) -> str:
         return "tasks"
 
-    def next(self, other: Task, operator: Operator = Operator.and_, on: Optional[TaskResult] = None) -> Task:
+    def next(
+        self,
+        other: Task,
+        operator: Optional[Operator] = None,
+        on: OnType = None,
+    ) -> Task:
         """Set self as a dependency of `other`."""
-        assert issubclass(other.__class__, Task)
-
-        condition = f".{on.value}" if on else ""
+        operator = operator or _default_next_operator.get()
+        on_list = _normalise_on(on, _default_next_on.get())
+
+        # Build condition string:
+        # - If multiple on-conditions: OR them and wrap in parens
+        # - If single on-condition: A.succeeded
+        # - If none: just "A"
+        if on_list and len(on_list) > 1:
+            condition_str = " || ".join(f"{self.name}.{c.value}" for c in on_list)
+            condition_str = f"({condition_str})"
+        elif on_list and len(on_list) == 1:
+            condition_str = f"{self.name}.{on_list[0].value}"
+        else:
+            condition_str = self.name
 
         if other.depends is None:
             # First dependency
-            other.depends = self.name + condition
+            other.depends = condition_str
         elif self.name in other._get_dependency_tasks():
             raise ValueError(f"{self.name} already in {other.name}'s depends: {other.depends}")
         else:
             # Add follow-up dependency
-            other.depends += f" {operator} {self.name + condition}"
+            other.depends += f" {operator} {condition_str}"
+
         return other
 
+    @classmethod
+    @contextmanager
+    def set_next_defaults(
+        cls,
+        operator: Optional[Operator] = None,
+        on: Union[TaskResult, Iterable[TaskResult], None] = None,
+    ):
+        """Temporarily modify the default behaviour of `next` and `>>`."""
+        on_list = _normalise_on(on)
+
+        old_operator = _default_next_operator.get()
+        old_on = _default_next_on.get()
+        _default_next_operator.set(operator or _default_next_operator.get())
+        _default_next_on.set(on_list)
+        try:
+            yield
+        finally:
+            _default_next_operator.set(old_operator)
+            _default_next_on.set(old_on)
```

Comment:
The preferred idiom is set/[reset](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar.reset)
```suggestion
        operator_token = _default_next_operator.set(operator or _default_next_operator.get())
        on_token = _default_next_on.set(on_list)
        try:
            yield
        finally:
            _default_next_operator.reset(operator_token)
            _default_next_on.reset(on_token)
```